### ISSUE TITLE
Summary Button: label identical to <nav> and <button>

### ISSUE THREAD

**Issue by cyberbaloo** [suitenumerique/docs#2031]
## Bug Report

**Problematic behavior**
"Sommaire" (Table of Contents) is indicated on both the `nav` element and the `button` to access it. The semantic roles of `nav` and `button` are not the same.

A screen reader will announce "Sommaire, navigation" and "Sommaire, button". This is confusing.

**Note that the opening of the menu is also not announced.**

**Expected behavior/code**
```
<nav id="summaryContainer" aria-label="Sommaire" class="sc-5b83f41b-0 fzTqLZ --docs--table-content">
    <div class="sc-5b83f41b-0 dOObJe">
        <div class="sc-5b83f41b-0 LRuEL">
            <span class="sc-5b83f41b-0 sc-458c56ea-0 hIpDzj fztac">Sommaire</span>
            <button aria-label="Sommaire" aria-expanded="true" aria-controls="toc-list" class="sc-5b83f41b-0 pVPoJ --docs--box-button ">
                <span aria-hidden="true" class="sc-5b83f41b-0 sc-458c56ea-0 ekowdF dUDhBq --docs--icon-bg material-icons">menu_open</span>
            </button>
```

**Possible Solution**
```
<nav aria-label="Sommaire" ...>
<button
aria-label="Afficher le sommaire"
aria-expanded="false"
aria-controls="toc-list"
...>
...
</button>
</nav>

```
**Additional context/Screenshots**
<img width="244" height="190" alt="Image" src="https://github.com/user-attachments/assets/cf84e205-5715-425d-9736-b383c6aa839d" />

**Critères RGAA**

Critère 7.1 : [Chaque script est-il, si nécessaire, compatible avec les technologies d'assistance ?](https://accessibilite.numerique.gouv.fr/methode/criteres-et-tests/#7.1)

**Impact** 
Blind or visually impaired users who use a screen reader will hear "Sommaire" (Table of Contents) twice in a row. It's difficult to understand that there is a region which is the navigation area and the other is a button to open the table of contents.
